### PR TITLE
Register allocation cleanup

### PR DESCRIPTION
**Pull request by kunalspathak**
While I was walking through the register allocator code, I saw few comments/code that was outdated.

### CODE REVIEW

**Comment by kunalspathak** [comment]
@dotnet/jit-contrib

**Review by CarolEidt — APPROVED** [review]
LGTM - thanks for the cleanup!

File: src/coreclr/src/jit/lsrabuild.cpp [hunk 1]
```diff
@@ -2266,7 +2263,7 @@ void LinearScan::buildIntervals()
                     {
                         JITDUMP("  Marking RP #%d of V%02u as spillAfter\n", interval->recentRefPosition->rpNum,
                                 interval->varNum);
-                        interval->recentRefPosition->spillAfter;
+                        interval->recentRefPosition->spillAfter = true;
```
Comment (sandreenko) [inline src/coreclr/src/jit/lsrabuild.cpp:2266]: Could it produce any diffs?

Comment (kunalspathak) [reply src/coreclr/src/jit/lsrabuild.cpp:2266]: That is a good question. May be I will validate that and post it here.

Comment (CarolEidt) [reply src/coreclr/src/jit/lsrabuild.cpp:2266]: Good catch, @sandreenko!

Comment (kunalspathak) [reply src/coreclr/src/jit/lsrabuild.cpp:2266]: Verified crossgen and PMI on framework libraries and I don't see any diffs.

**Pull request merged**